### PR TITLE
Add new builtin environment variables for triggering events

### PR DESCRIPTION
**Pull request by mohammedbabelly20**
Add the new builtin variables for triggering events `CM_TRIGGER_SOURCE` and `CM_BUILD_STARTED_BY`.

### CODE REVIEW

File: content/yaml-basic-configuration/environment-variables.md [hunk 1]
```diff
@@ -56,6 +56,8 @@ Here is a list of the built-in environment variables with brief explanations:
 | CM_ARTIFACT_LINKS       | Information about generated build artifacts that is available in post-publishing step. Read more about it below.                                                |
 | CM_WORKFLOW_NAME        | Name of the workflow that is currently being built       
 | CM_ENV                  | Environment file that sets/overrides environment variables at build time
+| CM_TRIGGER_SOURCE       | The source that triggered the build. Values are `webhook`, `schedule`, or `api`
```
Comment (helinanever) [inline content/yaml-basic-configuration/environment-variables.md:59]:
```suggestion
| CM_TRIGGER_SOURCE       | The source that triggered the build. Possible values are `webhook`, `schedule`, or `api`.
```
Wdyt of adding a brief explanation to each trigger source, e.g. `webhook` (initiated by a webhook event / Git webhook), `schedule` (triggered by a build schedule), or `api` (started manually from the UI or via the REST API).

File: content/yaml-basic-configuration/environment-variables.md [hunk 2]
```diff
@@ -56,6 +56,8 @@ Here is a list of the built-in environment variables with brief explanations:
...
+| CM_BUILD_STARTED_BY     | The email of the user who started the build if `CM_TRIGGER_SOURCE = api`
```
Comment (helinanever) [inline content/yaml-basic-configuration/environment-variables.md:60]: Maybe it's better to say "...the user who started the build from the UI".

Comment (mohammedbabelly20) [reply content/yaml-basic-configuration/environment-variables.md:60]: technically not just from UI, api call counts as well.
Since user is using the account's personal token.

File: content/yaml-basic-configuration/environment-variables.md (hunk 1 above)
Comment (mohammedbabelly20) [reply content/yaml-basic-configuration/environment-variables.md:59]: I don't know, I thing the names are self-explanatory 🤷🏻‍♂️
maybe yes, I can just explain the `api` value.

File: content/yaml-basic-configuration/environment-variables.md (hunk 2 above)
Comment (helinanever) [reply content/yaml-basic-configuration/environment-variables.md:60]: My point was that users must not know what the other environment variable holds. So sth like "The email address of the user that triggered the build. Available for builds started from the UI or via the REST API."

File: content/yaml-basic-configuration/environment-variables.md (hunk 1 above)
Comment (helinanever) [reply content/yaml-basic-configuration/environment-variables.md:59]: Ok, I agree about webhook and schedule. :)

File: content/yaml-basic-configuration/environment-variables.md (hunk 2 above)
Comment (mikhail-tokarev) [reply content/yaml-basic-configuration/environment-variables.md:60]: I'd probably rephrase it to start with when it's defined.
```
Used only with builds triggered by `api` (see `CM_TRIGGERED_SOURCE`). The email of the user who started the build.
```

**Review by helinanever — APPROVED** [review]
LGTM
